src/validation/Validator.ts

```diff
@@ -47,6 +47,34 @@ export class Validator {
             return executor.stripEmptyErrors(validationErrors);
         });
     }
+
+    /**
+     * Performs validation of the given object based on decorators used in given object class.
+     */
+    restrictValidate(object: Object, options?: ValidatorOptions): Promise<void>;
```

Comment:
bad name, we need to came up with better name, need to start name with `validate*`, and something like `validateOrThrow()` or better.